great_expectations/expectations/expectation.py

```diff
@@ -437,22 +470,23 @@ def _diagnostic_meta_properties_renderer(
         Here the custom column will be added in data docs.
         """
 
-        if result is None:
+        if not result:
             return []
         custom_property_values = []
-        meta_properties_to_render: Optional[dict]
-        if result.expectation_config is not None:
+        meta_properties_to_render: Optional[dict] = None
+        if result and result.expectation_config:
             meta_properties_to_render = result.expectation_config.kwargs.get(
                 "meta_properties_to_render"
             )
-        else:
-            meta_properties_to_render = None
-        if meta_properties_to_render is not None:
+        if meta_properties_to_render:
             for key in sorted(meta_properties_to_render.keys()):
                 meta_property = meta_properties_to_render[key]
-                if meta_property is not None:
+                if meta_property:
                     try:
                         # Allow complex structure with . usage
+                        assert isinstance(
+                            result.expectation_config, ExpectationConfiguration
+                        )
```

Comment:
Is this an invariant or is there the possibility that a user will face this?

If the latter, I say let's raise a `ValueError`

Reply:
✅